Sources/ZIPFoundation/FileManager+ZIP.swift

```diff
@@ -130,9 +130,7 @@ extension FileManager {
 
     func createParentDirectoryStructure(for url: URL) throws {
         let parentDirectoryURL = url.deletingLastPathComponent()
-        if !self.fileExists(atPath: parentDirectoryURL.path) {
-            try self.createDirectory(at: parentDirectoryURL, withIntermediateDirectories: true, attributes: nil)
-        }
+        try self.createDirectory(at: parentDirectoryURL, withIntermediateDirectories: true, attributes: nil)
```

Comment:
The `fileExists` check is there to silently return in case folder creation is not necessary. .

Reply:
I understand the intention, but, from the documentation:

"Attempting to predicate behavior based on the current state of the file system or a particular file on the file system is not recommended. Doing so can cause odd behavior or race conditions. It’s far better to attempt an operation (such as loading a file or creating a directory), check for errors, and handle those errors gracefully than it is to try to figure out ahead of time whether the operation will succeed."

Reply:
Does this make sense? From the docs, the return value is "true if the directory was created, true if createIntermediates is set and the directory already exists" so the check is not needed since if the directory already exists there will be no error.